### PR TITLE
RHEL 8 compatible executable

### PR DESCRIPTION
**Pull request by asaurer**
Build the executable in a way it can be executed on RHEL 8. Also, improve the build and release workflow.

### CODE REVIEW

**Comment by asaurer** [comment]
Please squash commits when merging

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,98 +8,53 @@ on:
...
-        ports: ["3306:3306"]
         options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
 
       database-tenant:
         image: mariadb:10.6
         env:
           MYSQL_DATABASE: ${{ env.MYSQL_DATABASE }}
           MYSQL_ROOT_PASSWORD: ${{ env.MYSQL_ROOT_PASSWORD }}
-        ports: ["3307:3306"]
+          MYSQL_TCP_PORT: 3307
```
Comment (mabiede) [inline .github/workflows/ci.yml:30]: Are these variables unused? -> remove

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+node_modules
+.parcel_cache
+.git
+_build
+_opam
+logs
+.devcontainer
+Dockerfile
```
Comment (mabiede) [inline .dockerignore:8]: Add an empty newline at the end.

- Also in Dockerfile.rhel8

File: docker/Dockerfile.rhel8 [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  && opam pin add -yn sihl-cache $repo_oxi/sihl.git \
+  && opam pin add -yn sihl-email $repo_oxi/sihl.git \
+  && opam pin add -yn sihl-queue $repo_oxi/sihl.git \
+  && opam pin add -yn sihl-storage $repo_oxi/sihl.git \ 
+  && opam pin add -yn sihl-token $repo_oxi/sihl.git \ 
+  && opam pin add -yn sihl-user $repo_oxi/sihl.git \
+  && opam pin add -yn conformist $repo_oxi/conformist.git \
+  && opam pin add -yn letters $repo_oxi/letters.git \
+  && opam pin add -yn canary $repo_uzh/canary.git \
+  && opam pin add -ywn guardian $repo_uzh/guardian.git \
```
Comment (mabiede) [inline docker/Dockerfile.rhel8:102]:
```suggestion
  && opam pin add -yn guardian $repo_uzh/guardian.git \
```

`-w` is for working directory. We should use it without this flag.

File: scripts/setup.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+opam pin add -yn sihl-queue $repo_oxi/sihl.git
+opam pin add -yn sihl-storage $repo_oxi/sihl.git
+opam pin add -yn sihl-token $repo_oxi/sihl.git
+opam pin add -yn sihl-user $repo_oxi/sihl.git
+opam pin add -yn conformist $repo_oxi/conformist.git
+opam pin add -yn letters $repo_oxi/letters.git
+
+# pin custom libraries
+opam pin add -yn canary $repo_uzh/canary.git
+opam pin add -ywn guardian $repo_uzh/guardian.git
```
Comment (mabiede) [inline scripts/setup.sh:26]:
```suggestion
opam pin add -yn guardian $repo_uzh/guardian.git
```

**Review by mabiede — CHANGES_REQUESTED** [review]

File: docker/Dockerfile.rhel8 (hunk 3 above)
Comment (asaurer) [reply docker/Dockerfile.rhel8:102]: this has been copied from the previous setup 🤷‍♂️

File: .github/workflows/ci.yml (hunk 1 above)
Comment (asaurer) [reply .github/workflows/ci.yml]: This was taken from the previous setup too. As I don't see a point in why the port has to be changed, I updated the code, so both services run on the default port.

**Review by mabiede — APPROVED** [review]

**Pull request merged**